### PR TITLE
[Messenger] Tell about messenger:consume invalid limit options

### LINKED ISSUE #48082: [Messenger]  Symfony\Component\Messenger\EventListener\StopWorkerOnTimeLimitListener::__construct(): Argument #1 ($timeLimitInSeconds) must be of type int, string given
### Symfony version(s) affected

5.4.14

### Description

Time limit being a non-int creates a problem for `messenger:consume`

### How to reproduce

```
bin/console messenger:consume -time-limit=5
```

Notice the typo.

### Possible Solution

Check the type or cast?

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #48082
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by derrabus** [comment]
Tests please. 🤓

**Comment by MatTheCat** [comment]
@derrabus okay, wasn’t sure because in the end an exception would be thrown anyway!

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php [hunk 1]
```diff
@@ -219,11 +228,11 @@ private function convertToBytes(string $memoryLimit): int
...
             case 't': $max *= 1024;
-            // no break
+                // no break
             case 'g': $max *= 1024;
-            // no break
+                // no break
             case 'm': $max *= 1024;
-            // no break
+                // no break
             case 'k': $max *= 1024;
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: Unrelated change, should be reverted.

**Comment by derrabus** [comment]
> in the end an exception would be thrown anyway!

Doesn't matter. If we feel like this issue is a bug is worth fixing, we should have tests that make sure it stays fixed.

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php (hunk 1 above)
Comment (MatTheCat) [reply src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: Fabbot complained about it. Should I ignore it?

Comment (derrabus) [reply src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: Yes. You didn't break it, you don't need to fix it.

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php [hunk 2]
```diff
@@ -163,7 +164,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         $stopsWhen = [];
-        if ($limit = $input->getOption('limit')) {
+        if (null !== ($limit = $input->getOption('limit'))) {
+            if (!is_numeric($limit) || 0 >= $limit) {
+                throw new InvalidOptionException(sprintf('Limit "%s" is not a positive integer.', $limit));
```
Comment (alamirault) [inline src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: In some places, we have this wording
 `throw new InvalidArgumentException(sprintf('Argument $maxItems must be a positive integer, %d passed.', $maxItems));`

```suggestion
                throw new InvalidOptionException(sprintf('Option "limit" must be a positive integer, "%s" passed.', $limit));
```

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php [hunk 3]
```diff
@@ -174,6 +179,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if (null !== ($timeLimit = $input->getOption('time-limit'))) {
+            if (!is_numeric($timeLimit) || 0 >= $limit) {
+                throw new InvalidOptionException(sprintf('Time limit "%s" is not a positive integer.', $timeLimit));
```
Comment (alamirault) [inline src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]:
```suggestion
                throw new InvalidOptionException(sprintf('Option "time-limit" must be a positive integer, "%s" passed.', $timeLimit));
```

File: src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php [hunk 4]
```diff
@@ -30,6 +31,10 @@ public function __construct(int $timeLimitInSeconds, LoggerInterface $logger = n
     {
         $this->timeLimitInSeconds = $timeLimitInSeconds;
         $this->logger = $logger;
+
+        if ($timeLimitInSeconds <= 0) {
+            throw new InvalidArgumentException('Time limit must be greater than zero.');
```
Comment (alamirault) [inline src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php:36]:
```suggestion
            throw new InvalidArgumentException(sprintf('Argument $timeLimitInSeconds must be a positive integer, %d passed.', $timeLimitInSeconds));
```

**Review by alamirault — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php (hunk 4 above)
Comment (MatTheCat) [reply src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php:36]: I got this one from https://github.com/symfony/symfony/blob/ff56fe80de77f4019d28cbe105ce1973b09b580a/src/Symfony/Component/Messenger/EventListener/StopWorkerOnMessageLimitListener.php#L35

**Review by chalasr — APPROVED** [review]

**Review by alamirault — APPROVED** [review]

File: src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php [hunk 5]
```diff
@@ -30,6 +31,10 @@ public function __construct(int $timeLimitInSeconds, LoggerInterface $logger = n
     {
         $this->timeLimitInSeconds = $timeLimitInSeconds;
         $this->logger = $logger;
+
+        if ($timeLimitInSeconds <= 0) {
```
Comment (welcoMattic) [inline src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php:35]:
```suggestion
        if (0 >= $timeLimitInSeconds) {
```

To be consistent with the 2 other if statements in your PR

**Review by welcoMattic — APPROVED** [review]

File: src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php (hunk 5 above)
Comment (MatTheCat) [reply src/Symfony/Component/Messenger/EventListener/StopWorkerOnTimeLimitListener.php:35]: I preferred to be consistent with https://github.com/symfony/symfony/blob/ff56fe80de77f4019d28cbe105ce1973b09b580a/src/Symfony/Component/Messenger/EventListener/StopWorkerOnMessageLimitListener.php#L34

(Like for https://github.com/symfony/symfony/pull/48085/files#r1012276670)

**Comment by chalasr** [comment]
Thank you @MatTheCat.

**Pull request merged**